rpc/src/calls/block.rs

```diff
@@ -74,6 +74,128 @@ where
+        Err(_) => {
+            return Err(Error::invalid_params(
+                "Takes [blockHash: DATA(64), full: BOOL]",
+            ));
+        }
+    };
+
+    let block_hash = match block_hash.get(2..) {
+        Some(bh) => String::from(bh),
+        None => {
+            return Err(Error::invalid_params("Invalid block hash, must have 0x"));
+        }
+    };
+
+    get_block_obj(BlockKey::Signature(block_hash), full, client)
+}
+
+/// Returns the number of transactions in a block matching the given block number.
+///
+/// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbynumber
+pub fn get_block_by_number<T>(params: Params, client: ValidatorClient<T>) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("eth_getBlockByNumber");
+
+    let (block_num, full): (String, bool) = match params.parse() {
+        Ok(t) => t,
+        Err(_) => {
+            return Err(Error::invalid_params(
+                "Takes [blockNum: QUANTITY|TAG, full: BOOL]",
+            ));
+        }
+    };
+
+    let block_key = match BlockKey::from_str(block_num.as_str()) {
+        Ok(k) => k,
+        Err(_) => {
+            return Err(Error::invalid_params("Invalid block number"));
+        }
+    };
+
+    get_block_obj(block_key, full, client)
+}
+
+/// Returns the number of transactions in a block from a block matching the given block hash
+///
+/// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblocktransactioncountbyhash
+pub fn get_block_transaction_count_by_hash<T>(
+    params: Params,
+    client: ValidatorClient<T>,
+) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("eth_getBlockTransactionCountByHash");
+
+    let (block_hash,): (String,) = match params.parse() {
+        Ok(t) => t,
+        Err(_) => {
+            return Err(Error::invalid_params("Takes [blockHash: DATA(64)]"));
+        }
+    };
+
+    let block_hash = match block_hash.get(2..) {
+        Some(bh) => String::from(bh),
+        None => {
+            return Err(Error::invalid_params("Invalid block hash, must have 0x"));
+        }
+    };
+
+    get_block_transaction_count(BlockKey::Signature(block_hash), client)
+}
+
+/// Returns the number of transactions in a block
+///
+/// Block is identified by number
+pub fn get_block_transaction_count_by_number<T>(
+    params: Params,
+    client: ValidatorClient<T>,
+) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("eth_getBlockTransactionCountByNumber");
+
+    let (block_num,): (String,) = match params.parse() {
+        Ok(t) => t,
+        Err(_) => {
+            return Err(Error::invalid_params("Takes [blockNum: QUANTITY|TAG]"));
+        }
+    };
+
+    let block_key = match BlockKey::from_str(block_num.as_str()) {
+        Ok(k) => k,
+        Err(_) => {
+            return Err(Error::invalid_params("Invalid block number"));
+        }
+    };
+
+    get_block_transaction_count(block_key, client)
+}
+
+//////////////////////
```

Comment:
This feels kind of unnecessary - don't all functions try their best to help?  The doc string for the function should indicate what it's for.